core/src/test/resources/org/apache/calcite/test/TopDownOptTest.xml

```diff
@@ -491,6 +491,81 @@ EnumerableMergeJoin(condition=[AND(=($1, $9), =($2, $10))], joinType=[inner])
 ]]>
     </Resource>
   </TestCase>
+    <TestCase name="testSortMergeJoinSubsetKey">
+        <Resource name="sql">
+            <![CDATA[select * from
+        sales.emp r join sales.bonus s on r.ename=s.ename and r.job=s.job
+        order by r.job desc nulls last]]>
+        </Resource>
+        <Resource name="planBefore">
+            <![CDATA[
+LogicalSort(sort0=[$2], dir0=[DESC-nulls-last])
+  LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8], ENAME0=[$9], JOB0=[$10], SAL0=[$11], COMM0=[$12])
+    LogicalJoin(condition=[AND(=($1, $9), =($2, $10))], joinType=[inner])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+      LogicalTableScan(table=[[CATALOG, SALES, BONUS]])
+]]>
+        </Resource>
+        <Resource name="planAfter">
+            <![CDATA[
+EnumerableMergeJoin(condition=[AND(=($1, $9), =($2, $10))], joinType=[inner])
+  EnumerableSort(sort0=[$2], sort1=[$1], dir0=[DESC-nulls-last], dir1=[ASC])
```

Comment:
Warning: the current implementation of EnumerableMergeJoin only supports keys sorted in ascending order nulls last, this will be a limitation if we ever want to actually run a plan like this one.

Reply:
Ack! Indeed it is. 

We have created a JIRA: https://issues.apache.org/jira/browse/CALCITE-4010. After this PR is done, I will ask Feng whether he still has time to work on CALCITE-4010. If not I can take it over to update the EnumerableMergeJoin implementation.

Reply:
I agree, this will be addressed by CALCITE-4010, I just wanted to make sure that we were aware of this current limitation